### DISCUSSION TITLE
Fan not running

### DISCUSSION THREAD

**Discussion by dev-zero** [cgomesu/nanopim4-satahat-fan#6] [Q&A] [answered]
Hi Carlos,

thanks a lot for the PWM script! Unfortunately it doesn't seem to work for me: no matter what I do, the fan (140mm, 3pin, when directly attached to 12V it works) is not spinning at all. Would you have ideas on where to start debugging?

Thanks a lot in advance,
Best,
Tiziano

**Comment by dev-zero** [comment]
To add more context, I'm on a 5.10.35-rockchip64 kernel from armbian.

**Comment by cgomesu** [comment]
hey, @dev-zero .  I noticed you are using a 140mm fan.  make sure your fan uses **less than 200mA** (.2A) because that's all the pwm fan pins on the sata hat can handle.  140mm fans often use more than that (e.g., [coolerguys 140mm fan](https://www.coolerguys.com/collections/140mm-fans/products/coolerguys-140mm-140x140x25-high-airflow-waterproof-ip67-fan)) and the pwm pins won't be able to handle it.  (I've actually never tested this myself.  this is an assumption based on the [sata hat documentation](https://wiki.friendlyarm.com/wiki/images/3/31/M4_SATA_HAT_Block.jpg).)

now, if that is not the problem, then I suspect that it might have something to do with the 5.* Kernel because I haven't noticed any issues with the pwm fan running on my NanoPi M4 (I run the legacy Kernel instead and its pretty reliable there).   here are a couple of things that I'd do then:

- [x] ~make sure it's not a hardware issue with the fan by using an external power source to turn it on~.
- [ ] post the output of `hostnamectl`
- [ ] manually run the `pwm-fan.sh` script and post the output here
- [ ] try running the fan manually from the sysfs. in the `pwm-fan.sh` file, check the `pwmchip`,  `export_pwmchip_channel`, and `fan_startup` functions for a reference. the relevant dir should be `/sys/class/pwm/pwmchip1/` (this is where the script will check by default) but it might have changed in the 5.* Kernel (??).  For one, I know that the `cpu` thermal sensor is now referred by `soc` instead, so other things might have changed as well.

Let me know how it goes.

**Reply by dev-zero** [reply]
Ok, I tried with a 125mA fan, still no luck. Not sure whether it was a coincidence or not, but when I had the fan attached during boot, I got PCIe timeouts and the SATA controller would not show up at all. After unplugging the fan and another reboot everything was fine again. Plugging the fan after the boot and running the script (with `-f`) didn't make it spin unfortunately. The base directory `.../pwmchip1/` is still there and I tried it manually before I discovered your script, still no luck.
Given that there are obviously electrical issues as well (the PCIe timeout) I'm likely going to replace the complete system by a Helios64 once it is available again.
Thanks again for your help!

**Comment by cgomesu** [comment, marked as answer]
this is unusual indeed.  if you have not lost hope yet, try the legacy
kernel instead. hardware-wise, you can debug electrical issues with a
multimeter.  in any case, the helios is a solid alternative but pricey and
supply is limited.
